### PR TITLE
feat!: pre-load CA certification chains on otaclient starts up, reject OTA if no CA certs are installed

### PR DESCRIPTION
**Pull request by Bodong-Yang**
## Introduction

This PR resolves an issue that comes from very old version of otaclient(since [v1.0.0](https://github.com/tier4/ota-client/tree/v1.0.0)), which results in OTA image sign cert verification being skipped when no valid CA cert chains can be imported by otaclient. Also the logic of loading and managing the CA cert chains are split away from `ota_metadata.legacy` package and becomes a standalone module in `ota_metadata.utils.cert_store`.

BREAKING CHANGE: Now otaclient will explicitly reject OTA when otaclient is not installed with valid CA cert chains. Also otaclient now only loads the CA cert chains once at startup. Re-configuring CA chains now requires otaclient restart.

## Tests

- [x] implement tests that cover the changes.
- [x] local pytest passed.
- [x] OTA test with VM passed. 
    - [x] OTA without CA installed will be rejected.
    - [x] OTA with CA installed, but with wrong sign.pem will be rejected.
    - [x] OTA with proper CA installed will be executed.

## Ticket
https://tier4.atlassian.net/browse/RT4-12647

### CODE REVIEW

**Comment by Bodong-Yang** [comment]
CA certs being loaded at otaclient starts up: 
```
[2024-10-23 15:52:10,063][INFO]-ota_metadata.utils.cert_store:load_ca_cert_chains:74,found installed CA chains: {'prd', 'dev', 'stg'}
[2024-10-23 15:52:10,065][INFO]-ota_metadata.utils.cert_store:load_ca_cert_chains:92,loaded CA cert chains: ['dev', 'prd', 'stg']
```

When no CA cert chain are found:
```
[2024-10-23 15:48:35,412][ERROR]-ota_metadata.utils.cert_store:load_ca_cert_chains:71,no CA cert chains found to be installed under the /opt/ota/client/certs!!!
[2024-10-23 15:48:35,412][ERROR]-otaclient.app.ota_client:__init__:625,failed to import ca_chains_store: CACertStoreInvalid('no CA cert chains found to be installed under the /opt/ota/client/certs!!!'), OTA will NOT occur on no CA chains installed!!!
```

**Comment by Bodong-Yang** [comment]
OTA rejected when no CA chains are imported:
```
[2024-10-23 15:48:50,690][INFO]-otaclient.app.ota_client:update:652,[update] entering local update...
[2024-10-23 15:48:50,692][ERROR]-otaclient.app.ota_client:_on_failure:642,
OTA failed with FAILURE: MetadataJWTVerficationFailed('no CA chains are installed, reject any OTA update')
```

**Comment by Bodong-Yang** [comment]
OTA rejected when failed to verify the OTA image sign cert: 
```
[2024-10-23 15:51:04,179][INFO]-ota_metadata.utils.cert_store:load_ca_cert_chains:74,found installed CA chains: {'stg', 'prd'}
[2024-10-23 15:51:04,182][INFO]-ota_metadata.utils.cert_store:load_ca_cert_chains:92,loaded CA cert chains: ['prd', 'stg']

...
[2024-10-23 15:51:07,542][INFO]-ota_metadata.legacy.parser:verify_metadata_cert:299,verify against prd failed: certificate signature failure
[2024-10-23 15:51:07,543][INFO]-ota_metadata.legacy.parser:verify_metadata_cert:299,verify against stg failed: certificate signature failure

...
@module: otaclient.app.ota_client
------ failure_reason ------
E309: unrecoverable OTA error, please contact technical support: certificate verification failed for OTA metadata.jwt
------ end of failure_reason -----
```

File: src/ota_metadata/legacy/parser.py [hunk 1]
```diff
@@ -267,46 +267,32 @@ def __init__(self, metadata_jwt: str, *, certs_dir: Union[str, Path]):
...
-            else:
-                raise MetadataJWTVerificationFailed("no pem file is found")
-        if len(ca_set_prefix) == 0:
-            logger.warning("there is no root or intermediate certificate")
-            return
-        logger.info(f"certs prefixes {ca_set_prefix}")
-
-        load_pem = partial(crypto.load_certificate, crypto.FILETYPE_PEM)
+        if not self.ca_chains_store:
+            _err_msg = "CA chains store is empty!!! immediately fail the verification"
```
Comment (Zhenfeng-Sun) [inline src/ota_metadata/legacy/parser.py:277]: Yes. This is the core change of this PR

**Review by Zhenfeng-Sun — APPROVED** [review]

**Pull request merged**